smithy-bindgen/tests/bindgen.rs

```diff
@@ -0,0 +1,48 @@
+
+
+#[test]
+fn test_str() {
+
+    mod test {
+        smithy_bindgen::smithy_bindgen!("keyvalue/keyvalue.smithy",
+             "org.wasmcloud.interface.keyvalue");
+    }
+
+    use test::StringList;
+
+    let _x = StringList::new();
+    println!("hello");
+ 
```

Comment:
Could we add a view more tests here that try to access some of the generated structs and traits? Could also just have it be a compile test by doing `use test::KeyvalueRequest as _;`

Reply:
That's what this test does - it accesses StringList, which is generated. I can add more structs from the same file, but the error conditions should be exactly the same as StringList.